### ISSUE TITLE
[Rec-IM] Hitting enter on a form should submit rather than cancel

### ISSUE THREAD

**Issue by amos-cha** [gordon-cs/gordon-360-ui#1807]
Any form component should be able to trigger the submit button (if the submit button is available) upon `enter` keypress

![Image](https://github.com/gordon-cs/gordon-360-ui/assets/78386128/4b0dd0c1-eac9-4c35-a51a-631bcd7713dd)

**Comment by amos-cha** [comment]
https://github.com/gordon-cs/gordon-360-ui/pull/1809
